src/App.js

```diff
@@ -52,9 +54,10 @@ const App = () => {
         <React.Fragment>
           <SimpleStoreDialog />
           <LoadingIndicator />
-          <InternetConnection />
-          {/* <ReCaptcha sitekey={Config.recaptcha} action="auth" verifyCallback={this.onRecaptcha} /> */}
-          {Splash}
+          <InternetConnection onDisconnect={<Splash />}>
```

Comment:
I think better will be pass the function which returns <Splash/>, in this case, we will not render splash in this component;

`onDisconnect={()=><Splash />}`

and in `InternetConnection` .   
`return disconnected ? props.onDisconnect &&  props.onDisconnect() : props.children`
@sirpy

Reply:
done